### DISCUSSION TITLE
Is there a `saved` event? for format on save.

### DISCUSSION THREAD

**Discussion by shirakami-chigusa** [Taitava/obsidian-shellcommands#419] [Q&A] [answered]
### Obsidian debug information

```
- Obsidian 1.6.5
- MacOS X Sonoma 14.5
- Shell Commands 0.22.0
```

### What shell are you using?

Zsh

### Please describe the problem

Is thre a `saved` event? I have confirmed that there is a `modified` event.
[File content modified - Shell commands documentation - Obsidian Publish](https://publish.obsidian.md/shellcommands/Events/File+content+modified)

I want to `format on save`. I have confirmed that it is possible to format on obsidian using shell command and `dprint`, so I would like to do this **when saving**.

set `Shell command withoud alias` to

```
dprint fmt {{file_name}}
```

[Markdown Plugin - dprint - Code Formatter](https://dprint.dev/plugins/markdown/)

**Comment by shirakami-chigusa** [comment, marked as answer]
Ah, I misunderstood how obsidian works.
Obsidian auto-saves the file even if I don't save it myself.

I now understand why there is no `saved` event.

I need to turn off autosave, but it's not a shell command thing, so I'll close this issue.

**Reply by Taitava** [reply]
Correct, there is no explicit `saved` event, as the `modified` event usually handles the same thing.

**Reply by shirakami-chigusa** [reply]
Thank you for the reply!
I assigned a hotkey to `shell commands: Execute dprint fmt {{file_name}}`. It's very useful.
